cwms_radar_api/src/main/java/cwms/radar/data/dao/LocationsDaoImpl.java

```diff
@@ -248,186 +250,129 @@ public static Feature buildFeatureFromAvLocRecord(Record avLocRecord) {
         }
 
-        Condition condition = buildCatalogWhere(unitSystem, office, idLike, categoryLike, groupLike)
-                .and(AV_LOC.LOCATION_ID.upper().greaterThan(locCursor));
-
-        SelectConditionStep<Record1<String>> tmp = dsl.selectDistinct(AV_LOC.LOCATION_ID)
-                .from(AV_LOC).innerJoin(AV_LOC_GRP_ASSGN).on(
-                        AV_LOC.LOCATION_ID.eq(AV_LOC_GRP_ASSGN.LOCATION_ID))
-                .where(condition);
-
-        Table<?> forLimit = tmp.orderBy(AV_LOC.LOCATION_ID).limit(pageSize).asTable();
-
-        SelectConditionStep<Record> query = dsl.select(
-                        AV_LOC.asterisk(),
-                        AV_LOC_GRP_ASSGN.asterisk()
-                )
-                .from(AV_LOC)
-                .innerJoin(forLimit).on(forLimit.field(AV_LOC.LOCATION_ID).eq(AV_LOC.LOCATION_ID))
-                .leftJoin(AV_LOC_GRP_ASSGN).on(AV_LOC_GRP_ASSGN.LOCATION_ID.eq(AV_LOC.LOCATION_ID))
-                .where(condition);
-
-        query.orderBy(AV_LOC.LOCATION_ID);
-//        logger.info( () -> query.getSQL(ParamType.INLINED));
-
-        Map<usace.cwms.db.jooq.codegen.tables.records.AV_LOC,
-                Set<LocationAlias>> theMap = new LinkedHashMap<>();
-
-
-        query.fetch().forEach(row -> {
-            usace.cwms.db.jooq.codegen.tables.records.AV_LOC loc = buildAvLoc(row);
-
-            Set<LocationAlias> locationAliases = theMap.computeIfAbsent(loc,
-                    k -> new LinkedHashSet<>());
-
-            locationAliases.add(buildLocationAlias(row));
-        });
-
-        List<? extends CatalogEntry> entries =
-                theMap.entrySet().stream()
-                        .sorted((left, right) -> (
-                                left.getKey().getLOCATION_ID()
-                                        .compareTo(right.getKey().getBASE_LOCATION_ID())
-                        ))
-                        .map(e -> buildCatalogEntry(e.getKey(), e.getValue())
-                        ).collect(Collectors.toList());
-
+        condition = condition.and(AV_LOC2.AV_LOC2.LOCATION_ID.upper().greaterThan(locCursor));
+
+        SelectLimitPercentStep<Record1<Long>> forLimit = dsl.select(AV_LOC2.AV_LOC2.LOCATION_CODE)
+            .from(AV_LOC2.AV_LOC2)
+            .where(condition.and(AV_LOC2.AV_LOC2.ALIASED_ITEM.isNull()))
+            .orderBy(AV_LOC2.AV_LOC2.LOCATION_ID)
+            .limit(pageSize);
+
+
+        SelectSeekStep2<Record, String, String> query = dsl.select(AV_LOC2.AV_LOC2.asterisk())
+            .from(AV_LOC2.AV_LOC2)
+            .where(condition)
+            .and(AV_LOC2.AV_LOC2.LOCATION_CODE.in(forLimit))
+            .orderBy(AV_LOC2.AV_LOC2.ALIASED_ITEM.desc(), AV_LOC2.AV_LOC2.LOCATION_ID);
+
+        List<? extends CatalogEntry> entries = query.fetch()
+            .stream()
+            .map(r -> r.into(AV_LOC2.AV_LOC2))
+            .collect(groupingBy(usace.cwms.db.jooq.codegen.tables.records.AV_LOC2::getLOCATION_CODE))
```

Comment:
As much as I love that particular operation does it exist in java 8? Though I might be getting that mixed up with an that works inline with the stream.

Reply:
yeah everything here is valid Java 8, I'm running and compiling in JDK 8 for everything RADAR.